apps/lottie-preview/src/configUtils.ts

```diff
@@ -19,59 +29,197 @@ export interface JsonField {
   originalEnabled: boolean;
 }
 
-export async function getJsonFields(cma: ConfigAppSDK['cma'], appDefinitionId: string): Promise<JsonField[]> {
-  const contentTypes = await cma.contentType.getMany({});
-
-  // Fetch all editor interfaces for contentTypes
-  const editorInterfacePromises = contentTypes.items.map((contentType) =>
-    cma.editorInterface.get({ contentTypeId: contentType.sys.id }).then((editorInterface) => ({ contentType, editorInterface }))
-  );
-
-  const resolvedEditorInterfaces = await Promise.all(editorInterfacePromises);
-  const jsonFields: JsonField[] = [];
-
-  for (const { contentType, editorInterface } of resolvedEditorInterfaces) {
-    for (const jsonField of contentType.fields.filter((f) => f.type === JsonFieldType)) {
-      const control = editorInterface.controls?.find((w) => w.fieldId === jsonField.id);
-      const isUsingApp = !!control && control.widgetId === appDefinitionId;
-      jsonFields.push({
-        contentTypeId: contentType.sys.id,
-        contentTypeName: contentType.name,
-        fieldId: jsonField.id,
-        fieldName: jsonField.name,
-        isEnabled: isUsingApp,
-        originalEnabled: isUsingApp,
-      });
+export interface JsonFieldsResult {
+  fields: JsonField[];
+  totalContentTypes: number;
+  processedContentTypes: number;
+  hasMore: boolean;
+}
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+// Helper function to retry API calls with exponential backoff
+async function retryWithBackoff<T>(fn: () => Promise<T>, retries: number = MAX_RETRIES, delay: number = RETRY_DELAY): Promise<T> {
+  try {
+    return await fn();
+  } catch (error: any) {
+    if (retries > 0 && (error.message?.includes('rate limit') || error.message?.includes('429'))) {
+      console.warn(`Rate limit hit, retrying in ${delay}ms... (${retries} retries left)`);
+      await sleep(delay);
+      return retryWithBackoff(fn, retries - 1, delay * 2); // Exponential backoff
     }
+    throw error;
+  }
+}
+
+// Add timeout wrapper for the entire operation
+const withTimeout = <T>(promise: Promise<T>, timeoutMs: number): Promise<T> => {
+  return Promise.race([promise, new Promise<T>((_, reject) => setTimeout(() => reject(new Error(`Operation timed out after ${timeoutMs}ms`)), timeoutMs))]);
+};
+
+export async function getContentTypesWithJsonFieldsCount(cma: ConfigAppSDK['cma']): Promise<number> {
+  try {
+    // Fetch ALL content types (lightweight, no editor interfaces)
+    const contentTypes = await cma.contentType.getMany({
+      query: { limit: 1000 },
+    });
+
+    let count = 0;
+    for (const contentType of contentTypes.items) {
+      const hasJsonFields = contentType.fields.some((f) => f.type === JsonFieldType);
+      if (hasJsonFields) {
+        count++;
+      }
+    }
+
+    return count;
+  } catch (error: any) {
+    console.error('getContentTypesWithJsonFieldsCount failed:', error.message);
```

Comment:
Are you sure this error will always have a `message` property? Some error objects don't and it could throw an error itself if it catches an error that doesn't have a `message` prop.  It might be safer to log the full `error` object in the second param instead of `error.message`